core/scrollbar.js

```diff
@@ -361,23 +431,51 @@ Blockly.Scrollbar.prototype.resizeHorizontal_ = function(hostMetrics) {
  * @private
  */
 Blockly.Scrollbar.prototype.resizeVertical_ = function(hostMetrics) {
-  var outerLength = hostMetrics.viewHeight - 1;
-  if (this.pair_) {
-    // Shorten the scrollbar to make room for the corner square.
-    outerLength -= Blockly.Scrollbar.scrollbarThickness;
-  } else {
+  this.resizeViewVertical_(hostMetrics);
+  this.resizeContentVertical_(hostMetrics);
```

Comment:
Bar length is the length of the scrollbar background; knobLength is the length of the scrollbar handle.  So bar length changes only when the view changes, but knobLength changes for both view and content changes.  Happy to take suggestions for better names, since I'm tunnel-visioned on the naming that was already here.

If resizeView is called, resizeContent also needs to be called.  But it should be possible to call resizeContent without resizeView.

Maybe that means that resizeView should call resizeContent, instead of relying on the caller to do it.  In that case, resize becomes a single call (to resizeView) for now, though it could then have some checks for whether resizeView needs to be called, or only resizeContent.

This is mostly a step toward being able to do them independently.